### ISSUE TITLE
nbdev_nb2md doesn't work in Colab due to CR/CRLF

### ISSUE THREAD

**Issue by IRailean** [fastai/nbdev#355]
When converting from .ipynb to .md cells with #hide are present in the markdown file.

Notebook:

![Untitled](https://user-images.githubusercontent.com/38012331/103283123-5efbe300-49e0-11eb-9846-0b24b9f7b770.png)

Markdown file contains:

# Title

```
#print("hello")
```

Test1

```
#hide 
print("test cell")
```

    test cell

Test2

**Comment by hamelsmu** [comment]
I'm not able to reproduce this behavior.  Steps I took:

1. Cloned `fastai/nbdev` repo
2. Made sure I am using the latest version of nbdev `pip install -e .`
3. navigated to notebook folder `cd nbs`
4. ran the following command `nbdev_nb2md 06_cli.ipynb`
5. when running `tail 06_cli.md` the last cell of the corresponding notebook is infact hidden.  

Closing this issue, but please do open a new issue that has a reproducible example with a real project & notebook that we can test against.  

cc: @jph00 

Thanks,

**Comment by IRailean** [comment]
@hamelsmu Thank you for the clarification. It works with Jupyter Notebooks indeed. The problem was that I`ve been using Colab Notebook. As I know in Colab nb there is additional metadata which makes it different from Jupyter nbs. Is nb2md supposed to work with Colab nbs?

**Comment by IRailean** [comment]
Colab adds to the end of some lines \r character. 
I created two identical notebooks, one in Jupyter Notebook, another in Colab. If you open them in text editor you will see, that in Jupyter it will look like:
'#hide\n'
and .ipynb from Colab is:
'#hide\r\n'

If you remove '\r' character and run nb2md all works ok.

**Comment by jph00** [comment]
@IRailean can you please provide a step-by-step way to reliably reproduce this issue please?

**Comment by IRailean** [comment]
Hi @jph00 . 
Here is how I reproduced the problem:
A.
1.Create a new notebook in Google Colab
2. Add 
```python
#hide
print("Hello world")
```
and run it
3. Create second separate Google Colab nb and run:
```python
!git clone https://github.com/fastai/nbdev.git
!pip install -e nbdev
```
to install nbdev
4. Run:
```python
import json
nb_path = '/content/drive/MyDrive/Colab Notebooks/Untitled7.ipynb'
with open(nb_path) as f:
  data = json.load(f)

print(data)
```
to get first notebook in json format (Untitled7 is the name of my first nb).
5. In the output find the content of the cell of the first nb.
In my case:
![image](https://user-images.githubusercontent.com/38012331/103467556-e20e9780-4d58-11eb-88e8-f97452ac2ce7.png)
 You can see that there are '\r\n' characters at the end of the line where '#hide' is located.

B.
I wanted to check if it is really problem in this extra '\r' and copy pasted some regex functions from nbdev  
(I've simplified 3 functions from nbdev, but didn`t change their`s core functionality)  
![image](https://user-images.githubusercontent.com/38012331/103467612-7da00800-4d59-11eb-92e1-42ab2c61f4d6.png)

And then if you run  
```python
remove_hidden(d['cells'][0])
```
There will be no match (#hide). But, if, first you rstrip the line and then run remove_hidden:  
```python
d['cells'][0]['source'] = d['cells'][0]['source'].rstrip('\r\n')
remove_hidden(d['cells'][0])
```
It fill find #hide in the line.

I've came up with the temporary solution: before running nbdev_nb2md I run:
```python
import json
nb_path = '/content/drive/MyDrive/Colab Notebooks/Untitled7.ipynb'
with open(nb_path) as f:
  data = json.load(f)

for cell in data['cells']:
  for idx, line in enumerate(cell['source']):
    cell['source'][idx] = cell['source'][idx].strip('\r\n') + '\n'
```
to remove '\r' character at the end of the line. (Yes, sometimes it will add an extra \n if there is no \r\n sequence, but as I am testing #hide functionality, it does not matter)

**Comment by hamelsmu** [comment]
@IRailean I am not sure nbdev explicitly supports development in Colab.  @muellerzr has done this but I know he encounters similar issues from time to time.  

Pinging @jph00 for his thoughts.

**Comment by muellerzr** [comment]
@hamelsmu I have a related PR here: https://github.com/fastai/nbdev/issues/363 I'll tackle the /r issue as well

**Comment by muellerzr** [comment]
@jph00 reopening for _now_. It seems that didn't 100% solve the issue, so I'm going to take a different approach to it. I say it didn't solve because the solution doesn't solve https://github.com/fastai/nbdev/issues/365 which has a similar issue (from what I can tell)

**Comment by hamelsmu** [comment]
@muellerzr friendly ping to see if you are still working on this or this can be closed?  Thank you

**Comment by hamelsmu** [comment]
Hi @muellerzr do you think that your fix in #366 could have caused https://github.com/fastai/nbdev/issues/422 ?

**Comment by muellerzr** [comment]
@hamelsmu yeah it should've been closed with that PR. That being said, yeah it caused #422. Will look there and continue it over there
